File: ainpc-core-plugin/src/main/java/ro/ainpc/listeners/ListenerRegistry.java
package ro.ainpc.listeners;

import org.bukkit.event.Listener;
import ro.ainpc.AINPCPlugin;
import ro.ainpc.gui.listeners.GuiInventoryListener;

/**
 * Punct unic pentru inregistrarea listener-elor pluginului.
 */
public class ListenerRegistry {

    private final AINPCPlugin plugin;

    public ListenerRegistry(AINPCPlugin plugin) {
        this.plugin = plugin;
    }

    public void registerAll() {
        register(new NPCInteractionListener(plugin));
        register(new NPCChatListener(plugin));
        register(new QuestObjectiveListener(plugin));
        register(new PlayerJoinListener(plugin));
        register(new VillagerLifecycleListener(plugin));
        register(new GuiInventoryListener(plugin));
    }

    private void register(Listener listener) {
        plugin.getServer().getPluginManager().registerEvents(listener, plugin);
    }
}

File: ainpc-core-plugin/bin/main/ro/ainpc/listeners/ListenerRegistry.kt
package ro.ainpc.listeners

import org.bukkit.event.Listener
import ro.ainpc.AINPCPlugin
import ro.ainpc.gui.listeners.GuiInventoryListener

/**
 * Punct unic pentru inregistrarea listener-elor pluginului.
 */
class ListenerRegistry(private val plugin: AINPCPlugin) {
    fun registerAll() {
        register(NPCInteractionListener(plugin))
        register(NPCChatListener(plugin))
        register(QuestObjectiveListener(plugin))
        register(PlayerJoinListener(plugin))
        register(VillagerLifecycleListener(plugin))
        register(MappingWandListener(plugin))
        register(GuiInventoryListener(plugin))
    }

    private fun register(listener: Listener) {
        plugin.server.pluginManager.registerEvents(listener, plugin)
    }
}
